### PR TITLE
[DDW-595] Fix lint warnings

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR refactors .js codebase with suggested lint fixes so we could clean all components from lint warnings.

## Todo:

- [x] Identify and fix broken tests
- [x] Finalize lint fixes for the remaining warnings
- [x] Retest everything

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by DeeJayElly** [comment]
> @DeeJayElly please add a CHANGELOG entry 🙏

Added

File: source/renderer/app/components/wallet/paper-wallet-certificate/SecuringPasswordDialog.js [hunk 1]
```diff
@@ -63,6 +63,16 @@ export default class SecuringPasswordDialog extends Component<Props, State> {
     securePasswordConfirmed: false,
   };
 
+  onSecurePasswordConfirmationChange = () => {
+    this.setState(prevState => ({
+      securePasswordConfirmed: !prevState.securePasswordConfirmed,
+    }));
+  };
+
+  onSecurePasswordConfirmationChange = this.onSecurePasswordConfirmationChange.bind(
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/paper-wallet-certificate/SecuringPasswordDialog.js]: @DeeJayElly please avoid declaring same functions twice (using the same name)

File: source/renderer/app/components/staking/StakingChart.js [hunk 2]
```diff
@@ -104,16 +104,20 @@ export default class StakingChart extends Component<Props, State> {
             minPointSize={2}
             isAnimationActive={false}
           >
-            {data.slice().map((entry, index) => {
+            {data.slice().map(entry => {
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/renderer/app/components/staking/StakingChart.js]: Done

File: source/renderer/app/components/sidebar/Sidebar.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ export default class Sidebar extends Component<Props> {
             const categoryClassName = kebabCase(category.name);
             return (
               <SidebarCategory
+                // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/renderer/app/components/sidebar/Sidebar.js]: Done

File: source/renderer/app/ThemeManager.js [hunk 4]
```diff
@@ -18,6 +19,7 @@ export default class ThemeManager extends Component {
     });
   }
   render() {
+    // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/renderer/app/ThemeManager.js]: // eslint-disable-next-line react/prop-types

Property types error

File: features/support/electron.js [hunk 5]
```diff
@@ -16,14 +16,19 @@ import {
...
 
+/* eslint-disable consistent-return */
+
 const context = {};
 const DEFAULT_TIMEOUT = 20000;
 let scenariosCount = 0;
 
 const printMainProcessLogs = () =>
   context.app.client.getMainProcessLogs().then(logs => {
+    // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply features/support/electron.js]: Done

File: source/main/ipc/index.js [hunk 6]
```diff
@@ -20,6 +20,7 @@ export default (window: BrowserWindow) => {
   getGpuStatus();
   handleBugReportRequests();
   handlePaperWalletRequests();
+  // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/main/ipc/index.js]: Done

File: source/renderer/app/components/wallet/paper-wallet-certificate/SecuringPasswordDialog.js (hunk 1 above)
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/paper-wallet-certificate/SecuringPasswordDialog.js]: Fixed

File: source/renderer/app/components/wallet/paper-wallet-certificate/PrintDialog.js [hunk 7]
```diff
@@ -82,6 +82,26 @@ export default class PrintDialog extends Component<Props, State> {
...
+
+  onConfirmReadable = () => {
+    this.setState(prevState => ({ isReadable: !prevState.isReadable }));
+  };
+
+  onConfirmScannable = () => {
+    this.setState(prevState => ({ isScannable: !prevState.isScannable }));
+  };
+
+  onConfirmReadable = this.onConfirmReadable.bind(this);
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/paper-wallet-certificate/PrintDialog.js]: Fixed

File: source/renderer/app/components/wallet/WalletSendForm.js [hunk 8]
```diff
@@ -151,6 +153,7 @@ export default class WalletSendForm extends Component<Props, State> {
   componentWillUnmount() {
     this._isMounted = false;
   }
+  submitOnEnter = submitOnEnter.bind(this, this.handleOnSubmit);
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/renderer/app/components/wallet/WalletSendForm.js]: Done

File: source/main/cardano/CardanoNode.js [hunk 9]
```diff
@@ -225,10 +227,10 @@ export class CardanoNode {
     const nodeCanBeStarted = await this._canBeStarted();
 
     if (!nodeCanBeStarted) {
-      return Promise.reject('CardanoNode: Cannot be started');
+      return Promise.reject(new Error('CardanoNode: Cannot be started'));
```
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply source/main/cardano/CardanoNode.js]: Done, all good there

File: features/support/electron.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (nikolaglumac) [reply features/support/electron.js]: What about this one @DeeJayElly ?

File: features/support/electron.js [hunk 10]
```diff
@@ -130,6 +135,7 @@ AfterAll(async function() {
     await printMainProcessLogs();
   }
   if (process.env.KEEP_APP_AFTER_TESTS === 'true') {
+    // eslint-disable-next-line
```
Comment (nikolaglumac) [inline features/support/electron.js]: @DeeJayElly we are missing details here too.

File: features/support/helpers/screenshot.js [hunk 11]
```diff
@@ -18,6 +18,7 @@ export const saveScreenshot = async (context, file) => {
     .capturePage()
     .then(imageBuffer => fs.writeFile(file, imageBuffer))
     .catch(err => {
+      // eslint-disable-next-line
```
Comment (nikolaglumac) [inline features/support/helpers/screenshot.js]: @DeeJayElly please add more details here

File: scripts/package.js [hunk 12]
```diff
@@ -59,6 +59,7 @@ if (version) {
 
 /** @desc Build, clear previous releases and pack new versions */
 async function startPack() {
+  // eslint-disable-next-line
```
Comment (nikolaglumac) [inline scripts/package.js]: @DeeJayElly please add more precise eslint rule here 🙏

File: scripts/package.js [hunk 13]
```diff
@@ -126,8 +127,9 @@ function pack(plat, arch, cb) {
  * @return {Function}
  */
 function log(plat, arch) {
-  return (err, filepath) => {
+  return (err) => {
     if (err) return console.error(err);
+    // eslint-disable-next-line
```
Comment (nikolaglumac) [inline scripts/package.js]: @DeeJayElly please add more precise eslint rule here 🙏

File: features/support/electron.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (DeeJayElly) [reply features/support/electron.js]: Done

File: features/support/helpers/screenshot.js (hunk 11 above)
Comment (DeeJayElly) [reply features/support/helpers/screenshot.js]: Done

File: scripts/package.js (hunk 12 above)
Comment (DeeJayElly) [reply scripts/package.js]: Done

File: scripts/package.js (hunk 13 above)
Comment (DeeJayElly) [reply scripts/package.js]: Done

File: features/support/electron.js (hunk 10 above)
Comment (DeeJayElly) [reply features/support/electron.js]: Done

**Pull request merged**